apis/r/cleanup

```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+rm -f  src/*.o src/*.so */*~ *~
```

Comment:
Do we need to delete shared objects on macos (`*.dylib`)?

Reply:
Yes we.  But as a non-user of macOS I don't habitually do that.  Is it still src/*.dylib or is it now src/*.so too?

Reply:
Looks like there's a mix of suffixes:

```
$ uname -a
Darwin ip-192-168-1-183.ec2.internal 21.6.0 Darwin Kernel Version 21.6.0: Mon Aug 22 20:17:10 PDT 2022; root:xnu-8020.140.49~2/RELEASE_X86_64 x86_64

$ pwd
/Users/johnkerl/git/single-cell-data/TileDB-SOMA

$ lsr \*.dylib
./build/libtiledbsoma/src/libtiledbsoma.dylib

$ lsr \*.so
./apis/python/src/tiledbsoma/libtiledbsoma.cpython-39-darwin.so
./dist/lib/libtiledbsoma.cpython-39-darwin.so
./build/libtiledbsoma/src/pyapi/libtiledbsoma.cpython-39-darwin.so
```

Reply:
That's good enough for me :)

```sh
edd@rob:~/git/tiledb-soma/apis/r(de/sc-22122/deduplicate)$ grep dylib ~/git/*/cleanup
/home/edd/git/anytime/cleanup:   src/*.dylib \
/home/edd/git/dtts/cleanup:rm -fr src/*.o src/*.so src/*.dll src/*.dylib *~ */*~ vignettes/auto/
/home/edd/git/gettz/cleanup:   src/*.dylib \
/home/edd/git/nanotime/cleanup:rm -fr src/*.o src/*.so src/*.dll src/*.dylib *~ */*~ vignettes/auto/
/home/edd/git/qlcal-r/cleanup:rm -f src/*.o src/*.so src/*.dylib src/*~ *~ \
/home/edd/git/rblpapi/cleanup:rm -rf src/*.o src/Rblpapi.so src/Rblpapi.dylib src/Rblpapi.dll src/Makevars \
/home/edd/git/rcppannoy/cleanup:rm -rf *~ */*~ src/*.o src/*.so src/*.dll src/*.dylib src/symbols.rds \
/home/edd/git/rcppapt/cleanup:rm -f src/*.o src/*.so src/*.dll src/*.dylib src/Makevars src/symbols.rds
/home/edd/git/rcppcctz/cleanup:rm -f src/*.o src/*.so src/*.dylib src/*~ *~ src/*.d
/home/edd/git/rcppclassic/cleanup:      inst/lib/libRcppClassic.so inst/lib/libRcppClassic.dylib \
/home/edd/git/rcppmlpack2/cleanup:rm -f src/*.o src/*.so src/*.dylib src/*~ *~
/home/edd/git/rcppqlcal/cleanup:rm -f src/*.o src/*.so src/*.dylib src/*~ *~ \
/home/edd/git/rcppquantuccia/cleanup:rm -f src/*.o src/*.so src/*.dylib src/*~ *~ \
/home/edd/git/rsqlite/cleanup:rm -f src/*.o src/*.so src/*.dll src/*.dylib
/home/edd/git/rvw/cleanup:rm -f src/Makevars src/*.o src/*.so src/*.dylib
edd@rob:~/git/tiledb-soma/apis/r(de/sc-22122/deduplicate)$ 
```

Reply:
> Looks like there's a mix of suffixes:

I think that is correct.  It used to be *.dylib, it may now now *.so but it won't hurt to add the `rm` so I just did.